File: python/day01/program.py
from pathlib import Path, PurePath


def get_top_one(list):
    list.sort()
    return list[-1]


def get_top_three(list):
    list.sort()
    return sum(list[-3::])


def transform_input(input):
    calories = []
    current_sum = 0

    for value in list(map(lambda x: 0 if x == "\n" else int(x), input)):
        if value == 0:
            calories.append(current_sum)
            current_sum = 0
            continue

        current_sum += value

    return calories


with open(Path.joinpath(Path(__file__).parent, "input.txt"), "r") as input:
    all_results = transform_input(input.readlines())
    part1_results = get_top_one(all_results)
    part2_results = get_top_three(all_results)

    print("Day 01")
    print("------")
    print(f"Part 1: {part1_results}")
    print(f"Part 2: {part2_results}")

File: ruby/day01/program.rb
lines = File.readlines(File.join(File.dirname(__FILE__), 'input.txt')).map(&:to_i)
calories = []
current = 0

lines.each do |l|
  current += l; next if l != 0

  calories << current
  current = 0
end

calories.sort! { |a, b| a <=> b }

pt1_result = calories.last
pt2_result = calories.last(3).sum

puts 'Day 1'
puts '-----'
puts "Pt. 1 result #{pt1_result}"
puts "Pt. 2 result #{pt2_result}"
